receiver/promtailreceiver/config.go

```diff
@@ -0,0 +1,117 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package promtailreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/promtailreceiver"
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/grafana/loki/clients/pkg/promtail/api"
+	"github.com/grafana/loki/clients/pkg/promtail/positions"
+	"github.com/grafana/loki/clients/pkg/promtail/scrapeconfig"
+	"github.com/grafana/loki/clients/pkg/promtail/targets/file"
+	"github.com/prometheus/client_golang/prometheus"
+	"go.opentelemetry.io/collector/confmap"
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v2"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/adapter"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/promtailreceiver/internal"
+)
+
+// PromtailConfig defines configuration for the promtail receiver
+type PromtailConfig struct {
+	InputConfig        Config `mapstructure:",squash" yaml:",inline"`
+	adapter.BaseConfig `mapstructure:",squash" yaml:",inline"`
+}
+
+// NewConfig creates a new input config with default values
+func NewConfig() *Config {
+	return NewConfigWithID(operatorType)
+}
+
+// NewConfigWithID creates a new input config with default values
+func NewConfigWithID(operatorID string) *Config {
+	return &Config{
+		InputConfig: helper.NewInputConfig(operatorID, operatorType),
+	}
+}
```

Comment:
Do these functions need to be public?

Reply:
`NewConfigWithID` should be public because it is called in `internal/components/receivers_test.go` https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/dde66116c4973842f19c3675e292f23d84966dfc/internal/components/receivers_test.go#L294
Made private other methods and struct you mentioned above. Thank you!

Reply:
Okay, let's leave that one public then